run-tests

```diff
@@ -19,7 +19,7 @@ FLAGS=(--verbose --no-keep-going)
 
 # To speed things up, we tell the C compiler to skip optimizations. (It's OK, the CI still uses -O2)
 # Also, add some compiler flags to verify standard compliance.
-export CFLAGS='-O0 -std=c99 -Wall -Werror -Wundef -Wpedantic -Wno-unused'
+export CFLAGS='-O0 -std=c99 -Wall -Werror -Wundef -Wno-unused'
```

Comment:
Are you sure we want to get rid of -Wpedantic? If I remember correctly, it's not just for C89. In this case it would know about c99 and warn about things that are not c99.

Reply:
From the GCC manual:

> -std=

> Determine the language standard.   This option is currently only supported when compiling C or C++.
> The compiler can accept several base standards, such as  c90  or  c++98,  and  GNU  dialects  of  those
           standards,  such  as  gnu90  or  gnu++98.   When a base standard is specified, the compiler accepts all
           programs following that standard plus those using GNU  extensions  that  do  not  contradict  it.   For
           example,  -std=c90  turns  off  certain features of GCC that are incompatible with ISO C90, such as the
           "asm" and "typeof" keywords, but not other GNU extensions that do not have a meaning in ISO  C90,  such
           as  omitting  the middle term of a "?:" expression. On the other hand, when a GNU dialect of a standard
           is specified, all features supported by the compiler are enabled, even when those features  change  the
           meaning  of  the  base  standard.   As  a result, some strict-conforming programs may be rejected.  **The
           particular standard is used by -Wpedantic to identify which features  are  GNU  extensions  given  that
           version  of  the  standard.  For example -std=gnu90 -Wpedantic warns about C++ style // comments, while
           -std=gnu99 -Wpedantic does not.**

Reply:
**From Wikipedia:**

ANSI C, ISO C, and Standard C are successive standards for the [C programming language](https://en.m.wikipedia.org/wiki/C_(programming_language)) published by the [American National Standards Institute](https://en.m.wikipedia.org/wiki/American_National_Standards_Institute) (ANSI) and [ISO/IEC JTC 1/SC 22](https://en.m.wikipedia.org/wiki/ISO/IEC_JTC_1/SC_22)/WG 14 of the [International Organization for Standardization](https://en.m.wikipedia.org/wiki/International_Organization_for_Standardization) (ISO) and the [International Electrotechnical Commission](https://en.m.wikipedia.org/wiki/International_Electrotechnical_Commission) (IEC). Historically, the names referred specifically to the original and best-supported version of the standard (known as C89 or C90). Software developers writing in C are encouraged to conform to the standards, as doing so helps [portability](https://en.m.wikipedia.org/wiki/Porting) between compilers.

**This is from GCC flags manual:**

-Wpedantic
-pedantic
Issue all the warnings demanded by strict ISO C and ISO C++; diagnose all programs that use forbidden extensions, and some other programs that do not follow ISO C and ISO C++. This follows the version of the ISO C or C++ standard specified by any -std option used.

Actually to be honest I am confused right now. From my experience, while using `-Wpedantic` with `-std=c11` I was getting warnings which was of `C89` standard as far as I can remember.

Reply:
I am bringing back `-Wpedantic` flag.

Here is where I was wrong: ISO C used to be historically and strictly referred to `C89`. But now it is not. ISO C standard in this case is the standard you specify using `-std=`.